x-pack/plugins/audit_trail/server/services/audit_trail_client.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Subject } from 'rxjs';
+import { KibanaRequest, Auditor, AuditableEvent } from 'src/core/server';
+import { AuditEvent } from '../types';
+
+import { SecurityPluginSetup } from '../../../security/server';
+import { SpacesPluginSetup } from '../../../spaces/server';
+
+interface Deps {
+  getCurrentUser: SecurityPluginSetup['authc']['getCurrentUser'];
+  getSpaceId: SpacesPluginSetup['spacesService']['getSpaceId'];
+}
+
+export class AuditTrailClient implements Auditor {
+  private readonly scope: string[] = [];
+  constructor(
+    private readonly request: KibanaRequest,
+    private readonly event$: Subject<AuditEvent>,
+    private readonly deps: Deps
+  ) {}
+
+  withScope = async <T>(name: string, fn: (...args: any[]) => Promise<T>): Promise<T> => {
```

Comment:
we either support async version only or use something similar to APM spon API:
```js
const span = apm.startSpan(...);
//...
if (span) span.end();
```

BTW should we rename it `withScope` --> `withSpan`?